javaslang/src/test/java/javaslang/collection/TreeSetTest.java

```diff
@@ -39,40 +38,29 @@ protected boolean emptyShouldBeSingleton() {
 
     @Override
     protected <T> TreeSet<T> of(T element) {
-        return TreeSet.of(toStringComparator(), element);
+        return TreeSet.of(naturalComparator(), element);
+    }
+
+    @Override
+    protected <T> TreeSet<T> of(Comparator<? super T> comparator, T element) {
+        return TreeSet.of(comparator, element);
     }
 
     @SuppressWarnings("varargs")
     @SafeVarargs
     @Override
     protected final <T> TreeSet<T> of(T... elements) {
-        boolean allLongs = true;
-        boolean allNumbers = true;
-        for (T element : elements) {
-            if (!(element instanceof Number)) {
-                allNumbers = false;
-            }
-            if (!(element instanceof Long)) {
-                allLongs = false;
-            }
-        }
-        if (allLongs) {
-            return TreeSet.ofAll(toLongComparator(), Iterator.of(elements));
-        } else if (allNumbers) {
-            return TreeSet.ofAll(toDoubleComparator(), Iterator.of(elements));
-        } else {
-            return TreeSet.ofAll(toStringComparator(), Iterator.of(elements));
-        }
+        return TreeSet.<T> of(naturalComparator(), elements);
```

Comment:
There is a way of overriding the default config by adding some files locally. There it should be possible to tweak it.

@pjagielski can you help?